packages/dara-core/tests/python/test_py_component.py

```diff
@@ -266,6 +266,59 @@ def TestBasicComp(input_val: int):
         assert response.json() == {'name': 'MockComponent', 'props': {'text': '3'}, 'uid': 'uid'}
         assert mock_func.call_count == 2
 
+async def test_derived_variables_with_df_nan():
+    """
+    Check that the py_component decorator returns a PyComponentDef when DerivedVariables are passed in and that it
+    processes DerivedVariables correctly when passed back in
+    """
+    import pandas
+    import numpy
+
+    builder = ConfigurationBuilder()
+
+    var = Variable()
+
+    # Test DerivedVariable can handle df with nan correctly
+    def func(var:pandas.DataFrame|None):
+        return pandas.DataFrame({'a':[1,2,numpy.nan]})
+
+    derived = DerivedVariable(func, variables=[var])
+
+    @py_component
+    def TestBasicComp(input_val: pandas.DataFrame):
+        return MockComponent(text=input_val.to_string())
```

Comment:
Does this actually run the resolver? Here you're explicitly `to_string`-ing the DataFrame so I don't think this will run the implicit serializer